source/DSCResources/MSFT_SccmIniFIle/MSFT_SccmIniFIle.psm1

```diff
@@ -0,0 +1,1107 @@
+        $SiteName,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SMSInstallDir,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SDKServer,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $PreRequisiteComp,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $PreRequisitePath,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $AdminConsole,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $JoinCeip,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $MobileDeviceLanguage,
+
+        [Parameter()]
+        [ValidateSet('EnforceHTTPS','HTTPorHTTPS')]
+        [String]
+        $RoleCommunicationProtocol,
+
+        [Parameter()]
+        [Boolean]
+        $ClientsUsePKICertificate,
+
+        [Parameter()]
+        [String]
+        $ManagementPoint,
+
+        [Parameter()]
+        [ValidateSet('HTTPS','HTTP')]
+        [String]
+        $ManagementPointProtocol,
+
+        [Parameter()]
+        [String]
+        $DistributionPoint,
+
+        [Parameter()]
+        [ValidateSet('HTTPS','HTTP')]
+        [String]
+        $DistributionPointProtocol,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $AddServerLanguages,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $AddClientLanguages,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $DeleteServerLanguages,
+
+        [Parameter()]
+        [ValidateSet('DEU','FRA','RUS','CHS','JPN','CHT','CSY','ESN','HUN','ITA','KOR','NLD','PLK','PTB','PTG','SVE','TRK','ZHH')]
+        [String]
+        $DeleteClientLanguages,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SQLServerName,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $DatabaseName,
+
+        [Parameter()]
+        [String]
+        $SqlSsbPort,
+
+        [Parameter()]
+        [String]
+        $SQLDataFilePath,
+
+        [Parameter()]
+        [String]
+        $SQLLogFilePath,
+
+        [Parameter()]
+        [String]
+        $CCARSiteServer,
+
+        [Parameter()]
+        [String]
+        $CasRetryInterval,
+
+        [Parameter()]
+        [ValidateRange(0, 100)]
+        [uint16]
+        $WaitForCasTimeout,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $CloudConnector,
+
+        [Parameter()]
+        [String]
+        $CloudConnectorServer,
+
+        [Parameter()]
+        [Boolean]
+        $UseProxy,
+
+        [Parameter()]
+        [String]
+        $ProxyName,
+
+        [Parameter()]
+        [String]
```

Comment:
same as proxy port Uint32?

Reply:
I was in the air about this. It get written to a file so it doesn't matter. I could make this a UInt16 and put a range so it has to be a valid port though. I used a UInt16 since it is valid for the whole port range and nothing more. So I don't need a validate set.
 Will do in my next push.